packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1897,5 +1952,70 @@ export function writeClientRenderBoundaryInstruction(
   }
 
   writeChunk(destination, boundaryID);
+  if (errorHash || errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorHash)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorHash)),
+      );
+  }
+  if (errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorMessage)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorMessage)),
+      );
+  }
+  if (errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    writeChunk(
+      destination,
+      stringToChunk(escapeJSStringsForInstructionScripts(errorComponentStack)),
+    );
+  }
   return writeChunkAndReturn(destination, clientRenderScript2);
 }
+
+const regexForJSStringsInScripts = /[<\'\`\u2028\u2029]/g;
+function escapeJSStringsForInstructionScripts(input: string): string {
+  if (typeof input !== 'string') {
```

Comment:
>(e.g. check the return value of `onRecoverableError`)

do you mean `onError`?

>It also means that the error can be more specific and we don't have to replicate it in other format configs.

I don't understand what you mean here.

>It would be better to check in inside ReactFizzServer when it first comes from user space code

yeah this makes sense to me